contracts/BatchExchange.sol

```diff
@@ -242,15 +243,14 @@ contract BatchExchange is EpochTokenLocker {
             "Claimed objective doesn't sufficiently improve current solution"
         );
         require(verifyAmountThreshold(prices), "At least one price lower than AMOUNT_MINIMUM");
-        require(tokenIdsForPrice[0] == 0, "fee token price has to be specified");
-        require(prices[0] == 1 ether, "fee token price must be 10^18");
+        require(tokenIdsForPrice[0] != 0, "Fee token has fixed price!");
         require(tokenIdsForPrice.checkPriceOrdering(), "prices are not ordered by tokenId");
         require(owners.length <= MAX_TOUCHED_ORDERS, "Solution exceeds MAX_TOUCHED_ORDERS");
         burnPreviousAuctionFees();
         undoCurrentSolution();
         updateCurrentPrices(prices, tokenIdsForPrice);
         delete latestSolution.trades;
-        int256[] memory tokenConservation = new int256[](prices.length);
+        int256[] memory tokenConservation = new int256[](prices.length + 1);
```

Comment:
I'd argue that the logic of initialising the data structure should be internal to tokenConversation. Is it possible to create an init method that takes the prices and returns an array?

The same applies to further below where we access the library internal datastructure directly:

```
uint256 burntFees = uint256(tokenConservation[0]) / 2;
```

I think this should also be a method on the library (such as `getFeesToBurn`). BatchExchange.sol should not have to know anything about the structure (e.g. if `feeToken` is stored on the 0th or last index) of the tokenConservation data type. This way a logic change like this will only require internal changes in the library.